### ISSUE TITLE
support for gatsby 4

### ISSUE THREAD

**Issue by amcc** [cloudinary-devs/gatsby-transformer-cloudinary#139]
Does this module support Gatsby v4, after upgrading a previously working site i'm finding inconsistent results with cloudinary images. We are using the plugin to use images on cloudinary and have a mongodb object with cloudinaryAssetData: true, this worked well for gatsby 3.

We see that nodes are transformed in the graphql on develop, they have the fixed, fluid objects present, however running develop is inconsistent, sometimes we see images, sometimes not. Build will have no images present. For the most part running develop has empty cloudinary nodes with only the very last having the src with a cloudinary URL.

It does look like the plugin hasn't been updated in a while, but if anyone can confirm whether it should run ok with gatsby 4 that would be useful and i can try to help debug whats going on here if there is an issue.

**Comment by djavco** [comment]
Adding to @amcc's comment I have tried a clean install of Gatsby4 with just the Mongo plugin and gatsby-transformer-cloudinary and with the same issue.

We are seeing all the allCloudinaryAsset nodes appearing in graphQL. Only a handful of the images are getting transformed correctly, before with lots of other plugins etc... it was only the very last image that was transformed.

We have had to customize the schema to ensure the cloudinary image object gets picked up by the plugin. See below where 'cloudinary' is the name of our image object on 'mongodbCollectionName'

```
 exports.createSchemaCustomization = ({ actions }) => {
   const { createTypes } = actions
   const typeDefs = `
     type mongodbCollectionName implements Node @infer {
       cloudinary: CloudinaryAsset @link(by: "id", from: "cloudinary___NODE")
     }
   `
   createTypes(typeDefs)
 }
```

**Comment by alistairlcc** [comment]
is there any progress on this issue?

**Comment by raae** [comment]
@olavea and I are tasked with getting this plugin up-to-date 🥳

We are looking into v4 compatibility as we speak. 
If you have any additional information to add, now would be great time to do so 🙏

**Comment by amcc** [comment]
@raae thanks for the update.

Perhaps one thing to roll in is this PR i made: https://github.com/cloudinary-devs/gatsby-transformer-cloudinary/pull/140
Its very simple, and we are using a patched version on our project - just a one liner to allow cloudinary nodes to be empty objects.

We are using mongodb as a backend, not sure that will be relevant to what you're doing, but it would obviously be wonderful that it works for us, i'm happy to test out any developments when you have them, just send me a branch or something. Also tell me if there's anything i should look at to help you out, right now we get no errors - just a bunch of null cloudinary nodes in graphql

**Comment by raae** [comment]
Could you share the minimal repo with me @amcc?

**Comment by amcc** [comment]
yep i can do that - probably can't give access to the db though, i'll create and grant you access asap

**Comment by raae** [comment]
The latter, then I can probably create a mongodb myself.

PS: I am pretty sure I figured out what is breaking.

**Comment by raae** [comment]
I managed to add examples to the demo site that breaks this so we'll work with that @amcc. But as soon as we have something testable I'll reach out.

**Comment by amcc** [comment]
sorry i've not got something back to you yet - swamped with a project launching - but much more free in a week or so, let me know if you still need something, or want something testing as thats very easy

**Comment by raae** [comment]
There will be a published beta tag soon for you to test 🎉

**Comment by raae** [comment]
Have you been able to the the gatsby-transformer-cloudinary@beta-v4 @amcc with your mongo db setup?

**Comment by cloudinary-raae** [comment]
Solved in v3.0.0
